code/datums/traits/negative.dm

```diff
@@ -671,3 +671,57 @@
 /datum/quirk/phobia/remove()
 	var/mob/living/carbon/human/H = quirk_holder
 	H.cure_trauma_type(/datum/brain_trauma/mild/phobia, TRAUMA_RESILIENCE_ABSOLUTE)
+
+/datum/quirk/british
+	name = "British"
+	desc = "You're British"
+	value = -1
+	gain_text = "<span class='danger'>Ye feel like a reet prat like, innit?.</span>"
+	lose_text = "<span class='notice'>You no longer feel like being rude and sassy.</span>"
+
+/datum/quirk/british/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	RegisterSignal(H, COMSIG_MOB_SAY, .proc/handle_speech)
+	var/datum/species/species = H.dna.species
+	var/liked = species.liked_food
+	species.liked_food = BEANS
+	species.disliked_food = liked
+
+/datum/quirk/british/remove()
+	var/mob/living/carbon/human/H = quirk_holder
+	UnregisterSignal(H, COMSIG_MOB_SAY)
+	var/datum/species/species = H.dna.species
+	species.liked_food = initial(species.liked_food)
+	species.disliked_food = initial(species.disliked_food)
+
+/datum/quirk/british/proc/handle_speech(datum/source, list/speech_args)
+	SIGNAL_HANDLER
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]"
+		var/list/whole_words = strings(BRIISH_TALK_FILE, "words")
+		var/list/british_sounds = strings(BRIISH_TALK_FILE, "sounds")
+		var/list/british_appends = strings(BRIISH_TALK_FILE, "appends")
+
+		for(var/key in whole_words)
+			var/value = whole_words[key]
+			if(islist(value))
+				value = pick(value)
+
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]")
+
+		for(var/key in british_sounds)
+			var/value = british_sounds[key]
+			if(islist(value))
+				value = pick(value)
+
+			message = replacetextEx(message, "[uppertext(key)]", "[uppertext(value)]")
+			message = replacetextEx(message, "[capitalize(key)]", "[capitalize(value)]")
+			message = replacetextEx(message, "[key]", "[value]")
+
+		if(prob(8))
+			message += pick(british_appends)
+	speech_args[SPEECH_MESSAGE] = trim(message)
```

Comment:
Clean up this code, use static lists when reading from huge text files.
Also make this a proc and don't have duplicated code